### PR TITLE
AMBARI-24833. Re-implement S3/HDFS outputs as global cloud outputs

### PR DESCRIPTION
**Pull request by oleewere**
# What changes were proposed in this pull request?
- ger rid of minio (reason: as it's really abstract solution for s3 backends, the policy handling is done by plain strings, not with objects)
- add hdfs client based and plain s3 outputs (as global configs)
- add hdfs output
- add a lot of new configs for s3 and hdfs
- delete s3 input/output, and hdfs output, delete log spooler framework as well

Cloud log process implementation:
- create specific loggers for every inputs
- rollover those logs into a different folder (archived)
- new thread which monitors the archive folders, if there is a new file, upload those to cloud storage
- we will have different configurations, to make sure logs are uploaded during shutdown (or even startup)

TODOs:
- load hadoop configs from file
- log/archive directories should not be in tmp folder, they should be configurable

No UTs yet
## How was this patch tested?
Manually with docker env + on local cluster. waiting UTs to pass

Please review @g-boros

### CODE REVIEW

File: ambari-logsearch-logfeeder/pom.xml [hunk 1]
```diff
@@ -145,9 +145,19 @@
...
-      <artifactId>minio</artifactId>
-      <version>5.0.1</version>
+      <groupId>org.apache.hadoop</groupId>
+      <artifactId>hadoop-aws</artifactId>
+      <version>3.1.1</version>
+    </dependency>
+    <dependency>
+      <groupId>com.amazonaws</groupId>
+      <artifactId>aws-java-sdk-core</artifactId>
+      <version>1.11.445</version>
```
Comment (swagle) [inline ambari-logsearch-logfeeder/pom.xml]: This should ideally be parameterized to help build scripts?

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/32/
Test FAILed.
Test FAILured.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/33/
Test PASSed.

File: ambari-logsearch-logfeeder/src/main/java/org/apache/ambari/logfeeder/output/cloud/CloudStorageUploader.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        Collection<File> filesToUpload = FileUtils.listFiles(new File(archiveLogDir), extensions, true);
+        if (filesToUpload.isEmpty()) {
+          logger.debug("Not found any files to upload.");
+        } else {
+          for (File file : filesToUpload) {
+            String basePath = uploadClient.getOutputConfig().getOutputBasePath();
+            String outputPath = String.format("%s/%s/%s/%s", clusterName, hostName, file.getParentFile().getName(), file.getName())
+              .replaceAll("//", "/");
+            logger.info("Upload will start: input: {}, output: {}", file.getAbsolutePath(), outputPath);
+            uploadClient.upload(file.getAbsolutePath(), outputPath, basePath);
```
Comment (swagle) [inline ambari-logsearch-logfeeder/src/main/java/org/apache/ambari/logfeeder/output/cloud/CloudStorageUploader.java]: Shouldn't the archives be deleted after upload? Or filter out ones already uploaded?

Comment (oleewere) [reply ambari-logsearch-logfeeder/src/main/java/org/apache/ambari/logfeeder/output/cloud/CloudStorageUploader.java]: that is handled by the uploadClient implementations, that is not extended to a method as with hdfs client that can be a parameter of the copy command

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/34/
Test PASSed.

File: ambari-logsearch-logfeeder/src/main/java/org/apache/ambari/logfeeder/output/cloud/CloudStorageOutput.java [hunk 3]
```diff
@@ -19,12 +19,139 @@
...
+  }
+
+  @Override
+  public String getShortDescription() {
+    return "cloud";
+  }
+
+  @Override
+  public String getStatMetricName() {
+    return null;
```
[the remark this answers is not in the document]
Comment (oleewere) [reply ambari-logsearch-logfeeder/src/main/java/org/apache/ambari/logfeeder/output/cloud/CloudStorageOutput.java]: null is handled

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/Ambari-LogSearch-Github-PR-Builder/37/
Test PASSed.

**Pull request merged**